modules/markup/html.go

```diff
@@ -87,13 +84,13 @@ func isLinkStr(link string) bool {
 	return validLinksPattern.MatchString(link)
 }
 
-// FIXME: This function is not concurrent safe
 func getIssueFullPattern() *regexp.Regexp {
-	if issueFullPattern == nil {
-		issueFullPattern = regexp.MustCompile(regexp.QuoteMeta(setting.AppURL) +
-			`\w+/\w+/(?:issues|pulls)/((?:\w{1,10}-)?[1-9][0-9]*)([\?|#]\S+.(\S+)?)?\b`)
-	}
-	return issueFullPattern
+	var issueFullPattern = regexp.MustCompile(regexp.QuoteMeta(setting.AppURL) +
+		`\w+/\w+/(?:issues|pulls)/((?:\w{1,10}-)?[1-9][0-9]*)([\?|#]\S+.(\S+)?)?\b`)
+
+	return func() *regexp.Regexp {
```

Comment:
Offtopic and I don't know go but would this really be needed (creating and calling this function)